dask_searchcv/methods.py

```diff
@@ -125,8 +125,20 @@ def _extract_pairwise(self, X, y, n, is_train=True):
 
 
 def cv_split(cv, X, y, groups, is_pairwise, cache):
-    check_consistent_length(X, y, groups)
-    return CVCache(list(cv.split(X, y, groups)), is_pairwise, cache)
+    if not cache or isinstance(cache, bool):
+        check_consistent_length(X, y, groups)
+        splits = list(cv.split(X, y, groups))
+        return CVCache(splits, is_pairwise, cache)
+    splits = list(cv.split(X, y, groups))
+    params = dict(pairwise=is_pairwise,
```

Comment:
Slightly shorter (and maybe clearer) to do

```python
cache.pairwise = pairwise
cache.cache = {}
cache.splits = splits
```

Is that equivalent to the `set_params` block?

Reply:
I'll change the `for` loop to your simple syntax and the `set_params` block to simplify in favor of just your snippet above, @TomAugspurger .  I don't think a `set_params`-style instance offers any advantage here.